server/server/data_loader/language_id.py

```diff
@@ -0,0 +1,75 @@
+import subprocess
+import langid
+import tempfile
+import random
+import pathlib
+import lxml.html
+
+def create_model(html_text_path, model_file, sample_size=20*1000*1000):
+    print('Creating langid model')
+    
+    with tempfile.TemporaryDirectory() as working_dir:    
+        corpus_dir = pathlib.Path(working_dir) / 'corpus'
+        domain_dir = corpus_dir / 'domain'
+        domain_dir.mkdir(parents=True)
+        
+        for lang_dir in html_text_path.glob('*'):
+            if not lang_dir.is_dir():
+                continue
+            lang_uid = lang_dir.name
+            if lang_uid == 'zz':
+                continue
+            domain_lang_dir = domain_dir / lang_uid
+            domain_lang_dir.mkdir()
+            files = sorted(lang_dir.glob('**/*.html'))
+            rand = random.Random(42)
```

Comment:
why 42?

Reply:
Well some kind of sampling is needed to make it faster to test, but it needs to be repeatable each time, sorting alone isn't good because then you'd only get `an` texts. So sorting + shuffle with predefined seed is a decent solution. And 42 is as good as any other seed.